### PR TITLE
Handle "create meeting" button

### PR DESCRIPTION
**Pull request by Chen-Ben-yashar**
hitting the "Meet!" button will open a popup with the top 3 suggestions.
If less were chosen (2 or 1) - will handle correctly.
if no slots were chosen - will inform user

once inside the popup, the system allows a single option to be chosen at a single time.

### CODE REVIEW

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java [hunk 1]
```diff
@@ -140,18 +140,21 @@ public int compare(Map.Entry<TimeSlot, Integer> o1,
...
         int iterationNumber = topSelectionToDisplay;
         if (iterationNumber > slotSelections.size()) {
             iterationNumber = slotSelections.size();
         }
         for (int i = 0; i < iterationNumber; i++) {
+            String currentSlot = "";
             TimeSlot topSlot = (TimeSlot) slotSelections.keySet().toArray()[i];
-            topSelections = String.format("%s %s %s - %d/%d \n", topSelections, topSlot.getDate(), topSlot.getHour(), getSelectionNumber(topSlot), membersAmount);
+            currentSlot = currentSlot + topSlot.getDate() + " " + topSlot.getHour() + " - " +
+                    + slotSelections.get(topSlot) + "/" + membersAmount + "\n";
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fix those lines in String.format()

File: app/src/main/java/com/example/meetapp/InsideGroupActivity.java [hunk 2]
```diff
@@ -108,13 +113,219 @@ public void run() {
...
+//                    @Override
+//                    public void onClick(View v) {
+//
+//                        currentButtonPressed =
+//                                handleOptionPress(new int[]{1, currentButtonPressed}, btnLst);
+//                    }
+//                });
+//
+//                topSuggestionsDialog.show();
+
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/InsideGroupActivity.java]: I really don't approve that big comment. it's not good practice to merge code in comments like that.
also it's too long. You need to divide all this code to functions.

File: app/src/main/java/com/example/meetapp/InsideGroupActivity.java [hunk 3]
```diff
@@ -108,13 +113,219 @@ public void run() {
...
+                // after sapir's fix
+//                option1.setText(stringTopSuggestionsArr.get(0));
+//                option2.setText(stringTopSuggestionsArr.get(1));
+//                option3.setText(stringTopSuggestionsArr.get(2));
+
+
+
+//                topSuggestionsDialog.show();
+
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/InsideGroupActivity.java]: same here with the code in comments and the function division.
also, what's Sapir's fix?

File: app/src/main/java/com/example/meetapp/InsideGroupActivity.java [hunk 4]
```diff
@@ -108,13 +113,219 @@ public void run() {
...
+////
+////
+////        makeGreen.setBackgroundResource(R.drawable.green_round_background);
+////        disable1.setBackgroundResource(R.drawable.custom_border);
+////        disable2.setBackgroundResource(R.drawable.custom_border);
+////        createMeeting.setBackgroundResource(R.drawable.green_round_background);
+////        isTopChoicePressed = true;
+//
+//    }
+
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/InsideGroupActivity.java]: same here with the code in comment and better function division. and remove all the empty lines.

File: app/src/main/java/com/example/meetapp/MenuHandler.java [hunk 5]
```diff
@@ -27,8 +27,10 @@
 import java.util.HashMap;
 import java.util.List;
 
-class MenuHandler {
-
+ class MenuHandler {
+    private final int NO_OPTION_CHOSEN = -1;
+    private int currentMeetingChoice = NO_OPTION_CHOSEN;
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/MenuHandler.java]: why the number here is -1?
also I think you don't need two variables here. In my opinion 
private int currentMeetingChoice = -1;
will do

File: app/src/main/java/com/example/meetapp/MenuHandler.java [hunk 6]
```diff
@@ -268,12 +272,162 @@ private void createTextView(String writeIn, int textSize, Context context){
...
+        // show only the relevant buttons
+        switch(numOfOptionsToDisplay)
+        {
+            case 1:
+                allOptionsLst[1].setVisibility(View.GONE);
+            case 2:
+                allOptionsLst[2].setVisibility(View.GONE);
+        }
+        return currentOptionLst;
+    }
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/MenuHandler.java]: function is too long.

File: app/src/main/java/com/example/meetapp/MenuHandler.java [hunk 7]
```diff
@@ -268,12 +272,162 @@ private void createTextView(String writeIn, int textSize, Context context){
...
+
+
+    private void setTextForOptions(Button[] AllOptionsLst)
+    {
+        // TODO: GET REAL CHOICES FROM SAPIR'S FUNCTION
+        AllOptionsLst[0].setText("#1:" + "Monday Morning");
+        AllOptionsLst[1].setText("#2:" + "Sunday Afternoon");
+        AllOptionsLst[2].setText("#3:" + "Saturday Evening");
+    }
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/MenuHandler.java]: I fixed the function in your branch when I solved the conflict. fix the code now to get real choices

File: app/src/main/java/com/example/meetapp/MenuHandler.java [hunk 8]
```diff
@@ -268,12 +272,162 @@ private void createTextView(String writeIn, int textSize, Context context){
...
+     private void handleExitBtnOfCreateMeetingPopup() {
+         TextView exitPopupBtn;
+         exitPopupBtn = topSuggestionsDialog.findViewById(R.id.exitPopupBtn);
+         exitPopupBtn.setOnClickListener(new View.OnClickListener() {
+             @Override
+             public void onClick(View v) {
+                 topSuggestionsDialog.dismiss();
+             }
+         });
+     }
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/MenuHandler.java]: theres a general exit popup handler "handleExitPopup". use it instead.

File: app/src/main/java/com/example/meetapp/MenuHandler.java [hunk 9]
```diff
@@ -268,12 +272,162 @@ private void createTextView(String writeIn, int textSize, Context context){
...
+         createMeeting.setOnClickListener(new View.OnClickListener() {
+             @Override
+             public void onClick(View v) {
+                 if(currentMeetingChoice != NO_OPTION_CHOSEN)
+                 {
+//                    cretaeMeetup();       //TODO: MEETUP IMPLEMENTATION HERE
+                     int currentRealChoice = currentMeetingChoice + 1;
+                     Toast.makeText(activity, "You Chose Option" + currentRealChoice, Toast.LENGTH_SHORT).show();
+
+                 }
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/MenuHandler.java]: delete empty line

File: app/src/main/java/com/example/meetapp/MenuHandler.java [hunk 10]
```diff
@@ -268,12 +272,162 @@ private void createTextView(String writeIn, int textSize, Context context){
...
+             }
+         });
+     }
+
+
```
Comment (sapirshp) [inline app/src/main/java/com/example/meetapp/MenuHandler.java]: same here, delete empty lines

File: app/src/main/res/layout/top_suggestions_popup.xml [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+            android:layout_height="wrap_content"
+            android:id="@+id/option3"
+            android:text="option3"
+            android:layout_margin="10dp"
+            android:background="@drawable/custom_border"
+
+            />
+
+
+
```
Comment (sapirshp) [inline app/src/main/res/layout/top_suggestions_popup.xml]: delete empty lines

File: app/src/main/res/layout/top_suggestions_popup.xml [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+            <!--android:id="@+id/CreateGroupBtn"-->
+            <!--android:layout_width="75dp"-->
+            <!--android:layout_height="75dp"-->
+            <!--android:layout_gravity="center_horizontal"-->
+            <!--android:layout_marginTop="20dp"-->
+            <!--android:layout_marginBottom="15dp"-->
+            <!--android:background="@drawable/disabled_button_background"-->
+            <!--android:text="@string/create"-->
+            <!--android:textColor="@color/blueDark"-->
+            <!--android:textSize="16sp" />-->
```
Comment (sapirshp) [inline app/src/main/res/layout/top_suggestions_popup.xml]: delete this comments

File: app/src/main/java/com/example/meetapp/InsideGroupActivity.java (hunk 2 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/InsideGroupActivity.java]: omg i just forgot to delete them... of course you are totally right!

File: app/src/main/java/com/example/meetapp/InsideGroupActivity.java (hunk 3 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/InsideGroupActivity.java]: same here.
sapir's function is "get top selections", i had a hard time getting the info correctly so i said i would leave it for later and use hard-coded days for now

File: app/src/main/res/layout/top_suggestions_popup.xml (hunk 12 above)
Comment (Chen-Ben-yashar) [reply app/src/main/res/layout/top_suggestions_popup.xml]: 👍

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 1 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: sapir will fix :)
thanks!

Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/MenuHandler.java (hunk 5 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: fine

File: app/src/main/java/com/example/meetapp/MenuHandler.java (hunk 6 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/MenuHandler.java (hunk 9 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/MenuHandler.java (hunk 10 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: fixed

File: app/src/main/res/layout/top_suggestions_popup.xml (hunk 11 above)
Comment (Chen-Ben-yashar) [reply app/src/main/res/layout/top_suggestions_popup.xml]: fixed

File: app/src/main/java/com/example/meetapp/MenuHandler.java (hunk 8 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: update your branch on your idea, theres a generic function there.
 private void handleExitPopup(final Dialog dialog, TextView exitBtn)
    {
        exitBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialog.dismiss();
                membersToAdd.clear();
            }
        });
    }

Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/MenuHandler.java (hunk 7 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: fixed it myself

**Pull request merged**